Library/Homebrew/migrator.rb

```diff
@@ -147,47 +169,51 @@ def pinned?
   end
 
   def migrate
-    if old_cellar.exist? && new_cellar.exist?
+    oh1 "Migrating #{Formatter.identifier(oldname)} to #{Formatter.identifier(newname)}"
+    lock
+    unlink_oldname
+    unlink_newname if new_cellar.exist?
+    repin
+    move_to_new_directory
+    link_oldname_cellar
+    link_oldname_opt
+    link_newname unless old_linked_keg.nil?
+    update_tabs
+  rescue Interrupt
+    ignore_interrupts { backup_oldname }
```

Comment:
Supposing we merging two `newname` and `oldname`, we'll end up with no `newname` at all if we `backup_oldname`.

Also, what If we remove `newname` kegs for merge to succeed?